examples/all-clusters-app/all-clusters-common/include/oven-operational-state-delegate.h

```diff
@@ -0,0 +1,116 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+#include <app/util/af-enums.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace OvenCavityOperationalState {
+
+// This is an application level delegate to handle operational state commands according to the specific business logic.
+class OvenCavityOperationalStateDelegate : public OperationalState::Delegate
+{
+private:
+    const Clusters::OperationalState::GenericOperationalState mOperationalStateList[8] = {
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kStopped)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kPaused)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kError)),
+    };
+
+    const Clusters::OperationalState::GenericOperationalPhase mOperationalPhaseList[1] = {
+        // Phase List is null
+        OperationalState::GenericOperationalPhase(DataModel::Nullable<CharSpan>()),
+    };
+
+public:
+    /**
+     * Get the countdown time. This attribute is not supported in our example app.
+     * @return Null.
+     */
+    DataModel::Nullable<uint32_t> GetCountdownTime() override { return {}; };
```

Comment:
Would be clearer as `return NullNullable;`, probably

Reply:
Updated per your suggestion to return NullNullable.